src/xmlcli/common/bios_fw_parser.py

```diff
@@ -242,11 +254,14 @@ def parse_binary(self, **kwargs):
     bios_size = file_size - buffer_pointer
     log.result(f"Size of BIOS: {bios_size} bytes ({bios_size // 1024} KB)")
 
-    self.output.update(self.parse_firmware_volume(buffer, buffer_pointer, end_point=file_size, bin_dir=bin_dir))
+    if self.isffs:
+      self.output.update(self.parse_ffs(buffer, buffer_pointer, end_point=file_size, par_tree_node=self.binary_tree, is_root_level=True, nesting_level=0, is_compressed=False, bin_dir=bin_dir))
+    else:
+      self.output.update(self.parse_firmware_volume(buffer, buffer_pointer, end_point=file_size, par_tree_node=self.binary_tree, is_root_level=True, bin_dir=bin_dir))
     log.result(self.stored_guids)
     return self.output
 
-  def parse_firmware_volume(self, buffer, buffer_pointer, end_point, nesting_level=0, is_compressed=False, **kwargs):
+  def parse_firmware_volume(self, buffer, buffer_pointer, end_point, par_tree_node=None, nesting_level=0, is_compressed=False, is_root_level=False, **kwargs):
```

Comment:
better to add new argument as suffix of function to maintain API function argument order